shared/src/commonMain/kotlin/io.newm.shared/internal/implementations/ConnectWalletUseCaseImpl.kt

```diff
@@ -1,37 +1,40 @@
 package io.newm.shared.internal.implementations
 
 import io.newm.shared.internal.repositories.CardanoWalletRepository
-import io.newm.shared.internal.repositories.ConnectWalletManager
 import io.newm.shared.internal.implementations.utilities.mapErrors
+import io.newm.shared.public.models.WalletConnection
 import io.newm.shared.public.usecases.ConnectWalletUseCase
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
 import shared.Notification
 import shared.postNotification
 
 internal class ConnectWalletUseCaseImpl(
-    private val connectWalletManager: ConnectWalletManager,
     private val cardanoWalletRepository: CardanoWalletRepository
 ) : ConnectWalletUseCase {
-    override fun connect(xpub: String) {
+    override fun connect(walletConnectionId: String) {
         mapErrors {
-            connectWalletManager.connect(xpub)
             postNotification(Notification.walletConnectionStateChanged)
+            cardanoWalletRepository.connectWallet(walletConnectionId)
```

Comment:
Should change it before sending the notification I think?
```suggestion
            cardanoWalletRepository.connectWallet(walletConnectionId)
            postNotification(Notification.walletConnectionStateChanged)
```